### PR TITLE
Update getDetailFromTab to consider pinnedLocation (AFTER site split)

### LINKED ISSUE #10241: Fix issue where pinned tab can't be really unpinned
## Test Plan
1. Go to feedly.com
2. Pin the tab
3. Bookmark it and change the location to https://feedly.com
4. Unpin the pinned tab
5. Relaunch Brave
6. There shouldn't be any pinned tab

## Original issue description
- Did you search for similar issues before submitting this one?
Yes
- Describe the issue you encountered:
Pinned tabs reappear next load after unpinning
- Platform (Win7, 8, 10? macOS? Linux distro?):
All
- Brave Version (revision SHA):
https://github.com/brave/browser-laptop/commit/526dc49875fc03c6b72f7d200939a05849c6ffca

- Steps to reproduce:
    1.  go to feedly.com
    2. pin the tab
    3. bookmark it and change the location to `https://feedly.com`
    4. unpin the pinned tab
    5. relaunch Brave
    
- Actual result:
There shouldn't be any pinned feedly tab
- Expected result:
The tab is still there

- Will the steps above reproduce in a fresh profile? If not what other info can be added?
Yes
- Is this an issue in the currently released version?
Yes
- Can this issue be consistently reproduced?
Yes
- Extra QA steps:
    1.
    2.
    3.

- Screenshot if needed:

- Any related issues:
https://github.com/brave/browser-laptop/issues/8477

cc @bsclifton

### PR DESCRIPTION
**Pull request by darkdh**
also normalize site key without trailing slash

fix #10241

Auditors: @bsclifton, @bbondy

Test Plan:
1. Go to feedly.com
2. Pin the tab
3. Bookmark it and change the location to https://feedly.com
4. Unpin the pinned tab
5. Relaunch Brave
6. There shouldn't be any pinned tab

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: js/state/siteUtil.js [hunk 1]
```diff
@@ -397,7 +400,7 @@ module.exports.removeSite = function (state, siteDetail, tag, reorder = true, sy
   state = siteCache.removeLocationSiteKey(state, location, key)
 
   const stateKey = ['sites', key]
-  let site = state.getIn(stateKey)
+  let site = state.getIn(['sites', key])
```
Comment (NejcZdovc) [inline js/state/siteUtil.js]: Is this change necessary? If so, can you remove line above?

Comment (darkdh) [inline js/state/siteUtil.js]: ah, I forgot to remove it. Thanks

Comment (NejcZdovc) [inline js/state/siteUtil.js]: np

**Comment by bsclifton** [comment]
Actually, sorry @darkdh- the above error is my fault. I had upgraded my session with the site splitting branch and this broke my session.  I'll try again with a proper session soon 😄

**Comment by bsclifton** [comment]
@darkdh can you please rebase this against master (since the split sites code was merged)?

I created a new PR for BEFORE that code gets merged (0.18 - 0.20) which you can find here 😄 
https://github.com/brave/browser-laptop/pull/10326

**Comment by bsclifton** [comment]
Closing in favor of https://github.com/brave/browser-laptop/pull/10326

This is not needed after the site split takes place
